config/initializers/redis.rb

```diff
@@ -9,14 +9,16 @@
   ENV['REDIS_URL'] = "redis://#{password.blank? ? '' : ":#{password}@"}#{host}:#{port}/#{db}"
 end
 
-Redis.current = Redis.new(
+namespace = ENV.fetch('REDIS_NAMESPACE') { '' }
```

Comment:
I don't have a 100% grasp on how subscriptions work but considering this is on the streaming service that means that when the upgrade happens and you change to a new REDIS_NAMESPACE that means all your previous subscriptions are now invalid since the id has the namespace prepended to it

So anyway I think that means you should be regenerating those subscriptions when you restart if a new REDIS_NAMESPACE is detected bc otherwise I think the streaming service will think it has valid subscriptions to <id> instead of <namespace>:<id>